FFMpegCore/FFMpeg/Pipes/RawAudioPipeSource.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace FFMpegCore.Pipes
+{
+    /// <summary>
+    /// Implementation of <see cref="IPipeSource"/> for a raw audio stream that is gathered from <see cref="IEnumerator{IAudioFrame}"/> 
+    /// </summary>
+    public class RawAudioPipeSource : IPipeSource
+    {
+        private readonly IEnumerator<IAudioSample> _sampleEnumerator;
+
+        public string Format { get; set; } = "s16le";
+        public uint SampleRate { get; set; } = 8000;
+        public uint Channels { get; set; } = 1;
+
+        public RawAudioPipeSource(IEnumerator<IAudioSample> sampleEnumerator)
+        {
+            _sampleEnumerator = sampleEnumerator;
+        }
+
+        public RawAudioPipeSource(IEnumerable<IAudioSample> sampleEnumerator)
+            : this(sampleEnumerator.GetEnumerator()) { }
+
+        public string GetStreamArguments()
+        {
+            return $"-f {Format} -ar {SampleRate} -ac {Channels}";
```

Comment:
I think it would be better to get sample format from `IAudioSample` rather then explicitly specify it in `RawAudioPipeSource`

Reply:
But what if the IAudioSamples in sampleEnumerator have different formats? I do see that your suggested change would only made this problem more obvious, not create it

Reply:
Indeed, this will probably be hard to ensure the enumerator is providing the same format of samples. A possibility would be to ensure the underlying byte array is the size as a "reference" sample (and as I'm writting that, I'm not even sure it would make sense at all...).

As from where the format is retrieves, I've put everything in the pipe configuration because as said above, I though it would make more sense for such abstract data to give the responsibility of data consistency to the user of the library. That said, both solutions sound fine, up to you to decide which one you prefer. 🙂

Reply:
I'm thinking that the current design, where format is specified on the pipesource, is way more straightforward to implement, and more importantly, to use. Maybe add a note about what is expected of the `sampleEnumerator` elements in a documentation comment